### PR TITLE
Add support for Data Removal API

### PR DESCRIPTION
**Pull request by balvig**
## What

Adds support for the new [Data Removal API](https://postmarkapp.com/developer/api/data-removals-api).

## How

Have some thoughts about the DRY-ness of these tests, but have gone for following local conventions for now!

### CODE REVIEW

File: spec/unit/postmark/account_api_client_spec.rb [hunk 1]
```diff
@@ -728,5 +728,46 @@
...
+        {
+          "ID" => 42,
+          "Status" => "Done"
+        }
+      }
+
+      it 'performs a POST request to /data-removals endpoint' do
+        allow(subject.http_client).to receive(:post).
+          with('data-removals', an_instance_of(String)).and_return(response)
+        subject.request_data_removal(:foo => 'bar')
```
Comment (ibalosh) [inline spec/unit/postmark/account_api_client_spec.rb:763]: wondering if we should use here more descriptive example data, like requested_by: .. etc. I know it's not mandatory here, but would make the test bit more descriptive, with having non random data for requesting removal.

File: spec/unit/postmark/account_api_client_spec.rb [hunk 2]
```diff
@@ -728,5 +728,46 @@
...
+        {
+          "ID" => 42,
+          "Status" => "Done"
+        }
+      }
+
+      it 'performs a GET request to /data-removals/:id endpoint' do
+        allow(subject.http_client).to receive(:get).
+          with('data-removals/42').and_return(response)
+        subject.get_data_removal_status(42)
```
[the remark this answers is not in the document]
Comment (balvig) [reply spec/unit/postmark/account_api_client_spec.rb:742]: Completely agree @ibalosh! 

Dunno about _"weird"_, but more likely to produce false positives for sure! 😓 

Was following local convention here, but look at following up with a PR switching all of these to `expect` instead.

File: spec/unit/postmark/account_api_client_spec.rb (hunk 1 above)
Comment (balvig) [reply spec/unit/postmark/account_api_client_spec.rb:763]: @ibalosh Yeh, this felt a little awkward to me too, but it seemed to be the done thing in this project, ie: 

https://github.com/ActiveCampaign/postmark-gem/blob/f4938b4d89629ead588485c54ee1623d5962a3a6/spec/unit/postmark/account_api_client_spec.rb#L624

I think these tests could do with a refresher at some point, but will stick with the local convention for now!

File: spec/unit/postmark/account_api_client_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ibalosh) [reply spec/unit/postmark/account_api_client_spec.rb:742]: nice!

**Pull request merged**